src/lib.rs

```diff
@@ -16,6 +16,74 @@
+//!
+//! The fundamental building block in Flue is the **process**: a concurrent
+//! execution thread with private memory. Processes may only communicate by
+//! passing **signals** between each other. Signals are sent to **routes** and
+//! received using **mailboxes**. All routes in a **route group** are killed if
+//! any of them are killed. The **post office** contains all of the routes in a
+//! set of communicating processes. **Capabilities** both reference and limit
+//! access to routes using fine-grained permission flags. **Tables** are stores
+//! of integer-addressed, unforgeable capabilities.
+//!
+//! Flue is made for the purpose of efficiently executing potentially untrusted
+//! process code, so to support running that code, Flue's security model has
+//! the following axioms:
+//! - Routes can only be accessed by a process if a capability to that route
+//!   has explicitly been passed to that process in a signal. You may sandbox
+//!   a process on a fine-grained level by limiting which capabilities get
+//!   passed to it.
+//! - Capabilities may only be created from other capabilities with either an
+//!   identical set or a subset of the original's permissions. Permission
+//!   escalation by untrusted processes is impossible.
+//!
+//! Here's a box diagram of two processes who each have a mailbox and a
+//! capability to the other's mailbox to help explain the mental model of a
+//! full Flue-based actor system:
+//!
+//! ```
+//!         Process A               Post Office               Process B
+//! ┌───────────────────────┐     ┌─────────────┐     ┌───────────────────────┐
+//! │         Table         │     │             │     │         Table         │
+//! │ ┌───────────────────┐ │     │ ┌─────────┐ │     │ ┌───────────────────┐ │
+//! │ │                   │ │     │ │         │ │     │ │                   │ │
+//! │ │ ┌───────────────┐ │ │  ┌──┼─► Route B ├─┼───┐ │ │ ┌───────────────┐ │ │
+//! │ │ │ Capability A  ├─┼─┼──┘  │ │         │ │   │ │ │ │ Capability B  ├─┼─┼─┐
+//! │ │ └───────────────┘ │ │     │ └─────────┘ │   │ │ │ └───────────────┘ │ │ │
+//! │ │                   │ │     │             │   │ │ │                   │ │ │
+//! │ └────────▲──────────┘ │     │ ┌─────────┐ │   │ │ └────────▲──────────┘ │ │
+//! │          │            │     │ │         │ │   │ │          │            │ │
+//! │   ┌──────┴───────┐    │  ┌──┼─┤ Route A ◄─┼─┐ │ │   ┌──────┴───────┐    │ │
+//! │   │   Mailbox A  ◄────┼──┘  │ │         │ │ │ └─┼───►   Mailbox B  │    │ │
+//! │   └──────────────┘    │     │ └─────────┘ │ │   │   └──────────────┘    │ │
+//! │                       │     │             │ │   │                       │ │
+//! └───────────────────────┘     └─────────────┘ │   └───────────────────────┘ │
+//!                                               │                             │
+//!                                               └─────────────────────────────┘
+//! ```
```

Comment:
It may actually be a good idea to make this a column or two skinnier if possible because its juuuuust a bit too long and it may wrap automatically on some peoples terminals

Reply:
i'll handle it